src/Summoner/Project.hs

```diff
@@ -54,7 +54,7 @@ generateProject projectName Config{..} = do
     settingsGitHub   <- decisionToBool cGitHub "GitHub integration"
     settingsTravis   <- ifGithub settingsGitHub "Travis CI integration" cTravis
     settingsAppVeyor <- ifGithub (settingsStack && settingsGitHub) "AppVeyor CI integration" cAppVey
-    settingsPrivat   <- ifGithub settingsGitHub "private repository" cPrivate
+    settingsPrivat   <- ifGithub (settingsGitHub && noUpload) "private repository" cPrivate
```

Comment:
I've tested it locally. Currently it asks about creating private project even I specify `--no-upload` option. That's why here should be negation of `noUpload` flag.
```suggestion
    settingsPrivat   <- ifGithub (settingsGitHub && not noUpload) "private repository" cPrivate
```